### DISCUSSION TITLE
Prediction deviation when training on  CP2K data

### DISCUSSION THREAD

**Discussion by rahulumrao** [deepmodeling/deepmd-kit#4486] [Q&A]
Hi,

I am currently training my model on CP2K data for a simple system. While the training process appears to converge, the predictions deviate significantly from the reference (CP2K) energies. 

![image](https://github.com/user-attachments/assets/8e2b6826-f436-4cde-a685-1b29b4b9cea6)

Interestingly, when I train the model on the data extracted from VASP output for same system, the parity plot looks fine.
![image](https://github.com/user-attachments/assets/8e16f245-2917-4bed-a9f4-583cfcca40b7)

I used same input file for training in both case.

[input.json](https://github.com/user-attachments/files/18222640/input.json)

I will appreciate any advice or suggestions!

Thank you in advance!

**Comment by njzjz** [comment]
I suggest checking if CP2k sets the same accuracy as VASP. It's also helpful to plot a figure for VASP vs CP2k.

**Reply by rahulumrao** [reply]
Thank you for the suggestion. However, I am not comparing the accuracy between models trained on two different MD packages. I am just trying to understand why the model training fails to converge when I use data from CP2K MD, while it converges quickly when I use VASP data for same system.

The model trained on CP2K data is unstable, as evident from the parity plot. LAMMPS MD crashes instantly when using this frozen model. Also, even if I increase the number steps for training the parity plot remains unchanged. 

My concern is why the training is not converging on CP2K data. Could there be some unit conversion, which I am missing?
I am using [CP2K plugin](https://github.com/robinzyb/cp2kdata/blob/main/docs/dpdata_plugin.md) from [dpdata](https://docs.deepmodeling.com/projects/dpdata/en/master/index.html) library to extract the data, so I assume the unit conversion is internally taken care of.

Below is the code snippet I used to extract the data from MD run :

```python
import dpdata
cp2kmd_dir = "."
cp2kmd_output_name = "job_cp2k.out"
dp = dpdata.LabeledSystem(cp2kmd_dir, cp2k_output_name=cp2kmd_output_name, fmt="cp2kdata/md")
print(dp)

Data Summary
Labeled System
-------------------
Frame Numbers      : 5001
Atom Numbers       : 6
Including Virials  : No
Element List       :
-------------------
C  H  Cl
1  3  2
```

**Reply by njzjz** [reply]
The concern is that your CP2K configuration didn't reach the same accuracy as that in VASP.

**Comment by rahulumrao** [comment]
If you are referring to the energy convergence of the CP2K configurations, they are well-converged. 

![image](https://github.com/user-attachments/assets/7a7d525f-b1a7-48e4-9195-7b08efa9d572)
